### DISCUSSION TITLE
Button Redirect to Subsection of Page e.g. setting/tab_index#invoices_specific_setting

### DISCUSSION THREAD

**Discussion by rossaddison** [yiisoft/docs#222] [General]
Is it possible to create a link e.g. setting/tab_index#invoices_specific_setting to go to a subsection of a page within a specific tab of a page?

So a page consists of a bootstrapped 5 tabbed index and I want to be redirected to a section within a specific tab that refers to a specific setting.

How will the hash i.e. '#' affect the arguments, the route, and the overall link? Is there any middleware involved?

**Comment by vjik** [comment]
You can use hash in URLs. It's shouldn't affects anything in middlewares and etc.

**Reply by rossaddison** [reply]
e.g working

http://invoice.myhost/invoice/setting/tab_index?active=invoices#settings[email_invoice_template]

The above url is working nicely. Just had to add an 'active' query to the settingController tab_index function i.e.

````
#[Query('active')] string $active = null,
````
By default any time the tab_index function is queried, the user will be directed to general but now the user can be directed to a specific setting that they need to fill in especially on exceptions and buttons on flash messages as in the example above. 

The specific setting in the view of course has to have an id e.g. id="settings[email_invoice_template]" which the hash or '#' then looks for.

**Reply by rossaddison** [reply]
I placed the following button at the bottom of inv/index.php to simulate it. 

````

<?php echo Html::a('test', $urlGenerator->generate('setting/tab_index', [], ['active' => 'invoices']).'#settings[email_invoice_template]')->render(); ?>
````
I had to concatenate the #settings[email_invoice_template] to the end of the $urlGenerator. Working fine. Thanks.

**Reply by vjik** [reply]
@rossaddison Your task is resolved? Can we close this discussion?

**Reply by rossaddison** [reply]
Yes thanks
